securedrop_client/gui/base/checkbox.py

```diff
@@ -0,0 +1,51 @@
+"""
+SecureDrop customized passphrase checkbox control
+
+A checkbox control created to toggle with hiding and showing PasswordEdit passphrases.
+Consists of a QCheckBox and a QLabel positioned horizontally within a QFrame.
+Present in the Sign-in and Export Dialog.
+"""
+from gettext import gettext as _
+
+from pkg_resources import resource_string
+from PyQt5.QtCore import Qt, pyqtSignal
+from PyQt5.QtGui import QCursor, QFont, QMouseEvent
+from PyQt5.QtWidgets import QCheckBox, QFrame, QHBoxLayout, QLabel, QWidget
+
+
+class SDCheckBox(QWidget):
+    clicked = pyqtSignal()
+    CHECKBOX_CSS = resource_string(__name__, "checkbox.css").decode("utf-8")
+    PASSPHRASE_LABEL_SPACING = 1
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.setObjectName("ShowPassphrase_widget")
+        self.setStyleSheet(self.CHECKBOX_CSS)
+
+        font = QFont()
+        font.setLetterSpacing(QFont.AbsoluteSpacing, self.PASSPHRASE_LABEL_SPACING)
+
+        self.layout = QHBoxLayout()
+        self.layout.setContentsMargins(0, 0, 0, 0)
+        self.layout.setSpacing(0)
+        self.setLayout(self.layout)
+
+        self.frame = QFrame()
+        self.frame.setLayout(QHBoxLayout())
+        self.frame.layout().setContentsMargins(0, 0, 0, 0)
+        self.frame.layout().setSpacing(0)
+
+        self.checkbox = QCheckBox()
+        self.label = QLabel(_("Show Passphrase"))
+        self.label.setFont(font)
+
+        self.layout.addWidget(self.frame)
+        self.frame.layout().addWidget(self.checkbox)
+        self.frame.layout().addWidget(self.label)
+        self.frame.setCursor(QCursor(Qt.PointingHandCursor))
+
+        self.clicked.connect(self.checkbox.click)
+
+    def mousePressEvent(self, e: QMouseEvent) -> None:
+        self.clicked.emit()
```

Comment:
This would trigger when the label is "right-clicked" or "middle-clicked", wouldn't it? Is that something we want? (It may be! I'm asking to be sure it's intentional :slightly_smiling_face:)

Reply:
Oops, sorry: let me clarify: the left click does trigger too. I don't _believe_ we want it to trigger on right or middle clicks, but I'd ask @creviera who's more familiar with the feature :slightly_smiling_face:

Reply:
> This would trigger when the label is "right-clicked" or "middle-clicked", wouldn't it? 

Yes, it does. Good catch.

I agree with the intuition that only a left-click should check and uncheck the checkbox. This seems to be pretty standard across desktop applications. Also, it's the default Qt behavior, so if you remove your method override, you'll see the desired behavior.

Reply:
Oh, also a tab + space bar should check and uncheck for a no-mouse experience. (It currently does this and will do this if you switch to default Qt behavior.)

Reply:
**Correction**: my intuition was wrong, and the native Qt checkbox gets activated by any mouse button press (not only when the left button is pressed). So this behavior is consistent with that. I'll resolve the comment. :white_check_mark: